packages/caver-wallet/src/keyringContainer.js

```diff
@@ -0,0 +1,308 @@
+        if (utils.isAddress(address)) {
+            keyringToRemove = this.getKeyring(address)
+        } else {
+            throw new Error(`To remove keyring, parameter should be address string.`)
+        }
+
+        if (keyringToRemove === undefined) return false
+
+        keyringToRemove.key = null
+        this._addressKeyringMap.delete(keyringToRemove.address.toLowerCase())
+
+        this._length--
+
+        return true
+    }
+
+    /**
+     * signs with data and returns the result object that includes `signature`, `message` and `messageHash`
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {string} data The data string to sign.
+     * @param {number} [role] A number indicating the role of the key. You can use `caver.wallet.keyring.role`.
+     * @param {number} [index] An index of key to use for signing.
+     * @return {object}
+     */
+    signMessage(address, data, role, index) {
+        const keyring = this.getKeyring(address)
+        if (keyring === undefined) throw new Error(`Failed to find keyring from wallet with ${address}`)
+        return keyring.signMessage(data, role, index)
+    }
+
+    /**
+     * signs the transaction using one key and return the transactionHash
+     *
+     * @param {string} address An address of keyring in keyringContainer.
+     * @param {Transaction} transaction A transaction object.
+     * @param {number} [index] An index of key to use for signing.
+     * @param {function} [hasher] A function to return hash of transaction.
+     * @return {string}
+     */
+    async signWithKey(address, transaction, index, hasher) {
+        if (!transaction.from || transaction.from === '0x') transaction.from = address
+        if (transaction.from.toLowerCase() !== address.toLowerCase())
+            throw new Error(`The from address of the transaction is different with the address of the keyring to use.`)
+
+        // Optional parameter processing
+        // (address transaction) / (address transaction index) / (address transaction hasher) / (address transaction index hasher)
+        if (_.isFunction(index) && hasher === undefined) {
+            hasher = index
+            index = 0
+        }
+        if (index === undefined) index = 0
+        if (!_.isNumber(index)) throw new Error(`Invalid index type: ${typeof index}`)
+        if (hasher === undefined) hasher = TransactionHasher.getHashForSigning
+
+        await transaction.fillTransaction()
+        const hash = hasher(transaction)
+        const role = transaction.type.includes('ACCOUNT_UPDATE') ? KEY_ROLE.ROLE_ACCOUNT_UPDATE_KEY : KEY_ROLE.ROLE_TRANSACTION_KEY
+
+        const keyring = this.getKeyring(address)
+        if (keyring === undefined) throw new Error(`Failed to find keyring from wallet with ${address}`)
+        const sig = keyring.signWithKey(hash, transaction.chainId, role, index)
+
+        transaction.appendSignatures(sig)
+
+        return hash
```

Comment:
Hmm.. Why do you return hash, not the transaction object?

Reply:
User already have transaction object so i think user doesn't need to receive duplicated object.

Reply:
Hmm.. It could be... What I'm wondering is that a user does not recognize or understand that the transaction object is modified in this function. For example, there are some functions where the passed parameter is unmodified and  the modified object is returned.

Reply:
Oh, i understand what you mean. 
Then i think return signed transaction object is better!!
I edited to return transaction object. 
And im using mock transaction in test code, so i can update caver.wallet.js test code after transaction implementation is merge. (for signed result detail check )

Reply:
Thanks. I have one more question related to this.
Can a user get the transaction hash using the returned object, or something else?

Reply:
user can get transaction hash through `tx.getTransactionHash()`